drivers/adc/adc_ite_it8xxx2.c

```diff
@@ -90,37 +95,50 @@ static int adc_it8xxx2_channel_setup(const struct device *dev,
 		LOG_ERR("Invalid channel reference");
 		return -EINVAL;
 	}
+	LOG_DBG("Channel setup succeeded!");
+
+	return 0;
+}
 
-	if (channel_cfg->channel_id < CHIP_ADC_CH4) {
+static void adc_enable_channel(uint32_t ch)
+{
+	if (ch < CHIP_ADC_CH4) {
 		/* For channel 0 ~ 3 control register. */
-		*adc_ctrl_regs[channel_cfg->channel_id].adc_ctrl =
-			(IT83XX_ADC_DATVAL | IT83XX_ADC_INTDVEN) +
-			channel_cfg->channel_id;
+		*adc_ctrl_regs[ch].adc_ctrl =
+			(IT8XXX2_ADC_DATVAL | IT8XXX2_ADC_INTDVEN) + ch;
 	} else {
 		/* For channel 4 ~ 7 control register. */
-		*adc_ctrl_regs[channel_cfg->channel_id].adc_ctrl =
-			IT83XX_ADC_DATVAL | IT83XX_ADC_INTDVEN | IT83XX_ADC_VCHEN;
+		*adc_ctrl_regs[ch].adc_ctrl =
+			IT8XXX2_ADC_DATVAL | IT8XXX2_ADC_INTDVEN |
+			IT8XXX2_ADC_VCHEN;
 	}
-	LOG_DBG("Channel setup succeeded!");
 
-	return 0;
+	/* Enable adc interrupt */
+	irq_enable(DT_INST_IRQN(0));
+
+	/* ADC module enable */
+	IT8XXX2_ADC_ADCCFG |= IT8XXX2_ADC_ADCEN;
 }
 
-static uint8_t count_channels(uint8_t ch)
+static void adc_disable_channel(uint32_t ch)
 {
-	uint8_t count = 0;
-	uint8_t bit;
-
-	bit = find_lsb_set(ch);
-	while (bit != 0) {
-		uint8_t idx = bit - 1;
-
-		ch &= ~BIT(idx);
-		bit = find_lsb_set(ch);
-		count++;
+	if (ch < CHIP_ADC_CH4) {
+		/*
+		 * For channel 0 ~ 3 control register.
+		 * bit(4:0) = 0x1f : channel disable
+		 */
+		*adc_ctrl_regs[ch].adc_ctrl =
+			IT8XXX2_ADC_DATVAL | 0x1F;
```

Comment:
Can you replace 0x1F with a macro: 
#define IT8XXX2_ADC_CHANNEL_DISABLED 0x1F

Reply:
Done